index.js

```diff
@@ -231,12 +226,31 @@ function getContentObject(stateResolveResultsObject, stateName) {
 	}, {})
 }
 
+function redirector(newStateName, parameters) {
+	return {
+		redirectTo: {
+			name: newStateName,
+			params: parameters
+		}
+	}
+}
+
 // { [stateName]: resolveResult }
 function resolveStates(states, parameters) {
 	var statesWithResolveFunctions = states.filter(isFunction('resolve'))
 	var stateNamesWithResolveFunctions = statesWithResolveFunctions.map(property('name'))
 	var resolves = Promise.all(statesWithResolveFunctions.map(function(state) {
-		return Promise.denodeify(state.resolve)(state.data, parameters)
+		return new Promise(function (resolve, reject) {
+			state.resolve(state.data, parameters, resolveCb, redirect)
```

Comment:
I'd like to leave the option open for people to return promises instead of calling the callback - how about copying the [pattern from denodeify](https://github.com/then/promise/blob/master/lib/node-extensions.js#L25-L28) to make that possible?